File: code/python/builder/ConcreteBuilder.py
from Builder import Builder
from Product import Product

class ConcreteBuilder(Builder):
    def __init__(self):
        self.product = Product()

    def build_part_a(self):
        self.product.add_part("Whole Wheat Bread")

    def build_part_b(self):
        self.product.add_part("Turkey")

    def build_part_c(self):
        self.product.add_part("Lettuce, Tomato, Onion")

    def get_product(self):
        return self.product

File: code/typescript/src/builder/ConcreteBuilder.ts
import { Builder } from "./Builder";
import { Product } from "./Product";

export class ConcreteBuilder implements Builder {
    private product: Product = new Product();

    public buildPartA(): void {
        this.product.addPart("Whole Wheat Bread");
    }

    public buildPartB(): void {
        this.product.addPart("Turkey");
    }

    public buildPartC(): void {
        this.product.addPart("Lettuce, Tomato, Onion");
    }

    public getProduct(): Product {
        return this.product;
    }
}
